### ISSUE TITLE
Licensing doubts

### ISSUE THREAD

**Issue by fpesari** [ComposersDesktop/CDP7#15]
I cannot figure the licensing for these files:

dev/new/dshift.c
dev/externals/portsf/ieee80.h
dev/externals/portsf/ieee80.c
dev/externals/paprogs/pvplay/pvfileio.h
dev/externals/paprogs/pvplay/pvdefs.h
dev/externals/paprogs/pvplay/pvfileio.c

They do not have the LGPL header nor any licensing info and they seem to have different authors from the rest of CDP7, so they possibly are copyrighted/proprietary files unless their developers signed a CLA.

This creates a problem for distributing CDP in some distro repositories, such as openSUSE and Debian.

**Comment by Sciss** [comment]
ieee80 seems public domain code: https://github.com/dpwe/dpwelib - origin should probably be documented

**Comment by jpffitch** [comment]
think that code i from MIT media lab in theearly 1990s

I have aversion that says

/***************************************************************\
*   ieee80.c                                                    *
*   Convert between "double" and IEEE 80 bit format             *
*   in machine independent manner.                              *
*   Assumes array of char is a continuous region of 8 bit frames*
*   Assumes (unsigned long) has 32 useable bits                 *
*   billg, dpwe @media.mit.edu                                  *
*   01aug91                                                     *
*   19aug91  aldel/dpwe  workaround top bit problem in Ultrix   *
*                        cc's double->ulong cast                *
*   05feb92  dpwe/billg  workaround top bit problem in          *
*                        THINKC4 + 68881 casting                *
\***************************************************************/

I hd i with coe that had an MIT licence.

Nt sure if that helps

**Comment by Sciss** [comment]
paprogs seems to have common ancestry with csound (LGPL) code: https://github.com/csound/csound/blob/develop/include/pvfileio.h

**Comment by fpesari** [comment]
> ieee80 seems public domain code: https://github.com/dpwe/dpwelib - origin should probably be documented

Yes, that would make life easier for packagers! Even a license header on both file with the original license notice would be fine :smile_cat:

> paprogs seems to have common ancestry with csound (LGPL) code: https://github.com/csound/csound/blob/develop/include/pvfileio.h

Nice catch, and I think the author of both is Richard Dobson, since `pvfileio.h` has the RWD initials in it. It'd be nice if we could get him to clear them.

So that would only leave `dev/new/dshift.c` without proper copyright info. Perhaps we should contact [Professor Rajmil Fischman at Keele University](https://www.keele.ac.uk/humanities/study/musicandmusictechnology/ourpeople/rajmilfischman/), the author of this file, to ask him if he can explicitly license it under the LGPL 2.1.

**Comment by Sciss** [comment]
dshift is an originary CDP algorithm I think: http://www.ensemble-software.net/CDPDocs/html/cgromody.htm#DSHIFT (but I'm not a CDP author, so someone from the project can probably clarify this contribution)

**Comment by musger** [comment]
This issue might be resolved by 8d0f9d1 and can possibly be closed?

**Comment by fpesari** [comment]
> This issue might be resolved by [8d0f9d1](https://github.com/ComposersDesktop/CDP7/commit/8d0f9d11a055fc1fdca181f2a7c1a06e39844f4b) and can possibly be closed?

But I don't understand, the comment says the author is Rajmil Fischman and the license notice says it is @richarddobson...perhaps he can weigh in :)

**Comment by jpffitch** [comment]
I cannot see anything wrong or odd about the dshift source file, and in 
particular no reference to dobson.  Can you explain?

**Comment by fpesari** [comment]
> I cannot see anything wrong or odd about the dshift source file, and in particular no reference to dobson. Can you explain?

Before commit https://github.com/ComposersDesktop/CDP7/commit/8d0f9d11a055fc1fdca181f2a7c1a06e39844f4b , dshift had no licensing info and a copyright attribution to prof. Rajmil Fischman at Keele University (it's still there). 

Unless prof. Fischman signed a Contributor Licensing Agreement which gives copyright to prof. Dobson (in that case, the current licensing notice would be correct), that file cannot be under the LGPL.

I think CDP can be built without `dshift` so this is not really a problem for my purpose (shipping an openSUSE package, which requires an OSI-compliant license for all files), but I think it'd be great for CDP7's inclusion in many distro repositories if these doubts were completely cleared.

**Comment by jpffitch** [comment]
I still d not follow.  I am invoved with another audio sofware system and 
we have copyright notices from tens of different authors and a very 
similar licence statement for LPGL; no one has queried this and it is 
packaged by Debian.  After that commit I cannot see the difference.

**Comment by fpesari** [comment]
> I still d not follow. I am invoved with another audio sofware system and we have copyright notices from tens of different authors and a very similar licence statement for LPGL; no one has queried this and it is packaged by Debian. After that commit I cannot see the difference.

The problem is not the LGPL, it's the fact that CDP was originally released under a proprietary license. In order to relicense to LGPL, AFAIK ([Source 1](https://en.wikipedia.org/wiki/Software_relicensing#Motivation_and_description), [Source 2](https://www-archive.mozilla.org/mpl/relicensing-faq#no-license-notice)) permission is needed from all authors except if they signed a CLA (a clause that says for example, "all code belongs to the Composers Desktop Project").

**Comment by jpffitch** [comment]
Two things.

One Rajmil has agreed to the licence change

Two my other audio software changed from a restrictive licence to LGPL and I collected agreements but the copyright remains with the author.



⁣Sent from TypeApp ​

**Comment by fpesari** [comment]
> Two things. One Rajmil has agreed to the licence change

Well, if that's the case then case closed! But we outsiders to the CDP project had no way of knowing :)
